rpki-validator/src/main/java/net/ripe/rpki/validator3/util/HappyEyeballsResolver.java

```diff
@@ -237,7 +237,7 @@ private boolean openConnectionFromQueue(Queue<InetAddress> queue) throws Interru
     }
 
     private static boolean queueExhausted(Queue<InetAddress> queue) {
-        return !queue.isEmpty() && queue.peek() == ADDRESSES_TERMINATOR;
+        return queue.peek() == ADDRESSES_TERMINATOR;
```

Comment:
As a suggestion: do you want to have Queue<Optional<InetAddress>> and use 
queue.isEmpty() && !queue.peek().isPresent(), so you don't have to have a specific NULL objects?

Reply:
Yes, I think I like this idea…